# Eletrônica de Potência II - Tarefa 1

Alunos:
* Alejo Perdomo Milar
* João Mário Carnieletto Izoton Lago

# Conversor Buck com seguintes especificações:

* Tensão de Entrada: 36 V
* Tensão de Saída: 12 V
* Freq. de comutação: 50 kHz;
* Indutância L: 100 uH
* Potência: 100 W;

# Simulação preliminar usando capacitor estimado de 100uF

Para ter uma ideia de valores para escolha do capacitor, vamos simular o circuito com um capacitor de 100uF.

A imagem a seguir apresenta o circuito simulado no software LTSpice, utilizando chaves ideais.

![Schematic](./prints/schematic_1.png)

---

## Formas de Onda Típicas

### Ondas na carga com capacitor de 100uF

![typical](./prints/power_1.png)
![typical](./prints/current_1.png)
![typical](./prints/voltage_1.png)

---

### Delta de corrente no indutor para obter o capacitor para o ripple desejado

![typical](./prints/current_inductor_1.png)

---

# Dimensionamento do capacitor para ripple de saída de 1%

Assumindo um delta de tensão de 1% da tensão de saída, temos:

$$
\Delta V_o = 0.01 \cdot V_{out} = 0.01 \cdot 12 V = 0.12 V
$$

A corrente média no capacitor é dada por:

$$
C_o = \frac{\Delta I_L}{8 \cdot f \cdot \Delta V_o}
$$

Dados os parâmetros do conversor, temos:

* Delta de corrente indutor: 1.5754 A
* Delta de tensão: 0.12 V
* Frequência de comutação: 50 kHz

Substituindo os valores na equação, temos:

$$
C_o = \frac{1.5754 A}{8 \cdot 50 kHz \cdot 0.12 V} = 32.820 uF
$$


# Simulação usando capacitor calculado 33uF

A imagem a seguir apresenta o circuito simulado no software LTSpice, utilizando chaves ideais.

![Schematic](./prints/schematic_2.png)

---

## Formas de Onda Típicas

### Ondas na carga com capacitor de 100uF

![typical](./prints/power_2.png)
![typical](./prints/current_2.png)
![typical](./prints/voltage_2.png)

> Como pode ser visto na imagem, o ripple de tensão na carga é de 0.12V conforme desejado

---

### Correntes no capacitor

![typical](./prints/current_capacitor_2.png)
![typical](./prints/current_capacitor_2_2.png)

Conforme as ondas pode se extrair que o capacitor tem uma corrente RMS $462 mA$ e um delta de corrente de $1.6 A$. Esses valores são utilizados para escolha do componente.

## Capacitor escolhido 5 x (CL32B106KBJNFNE)

Foi escolhido o uso de 5 unidades dos capacitores [CL32B106KBJNFNE](https://br.mouser.com/ProductDetail/Samsung-Electro-Mechanics/CL32B106KBJNFNE?qs=%252B6g0mu59x7JyYu1TrhK7vQ%3D%3D&srsltid=AfmBOoqsUTMLkSxsrZveaIdgTWPKhmQ63iqWpHIKn3tUCvS0Xo5BivaT) de 10 uF cada devido ao fato de serem cerâmicos é necessário fazer uma compensação de queda de capacitância com a tensão aplicada.

Assim sendo tem de se analisar 2 fatores:

### Perda de capacitância com a tensão aplicada

![typical](./datasheet/CL32B106KBJNFNE_cap_loss.png)

Dado que aplica-se 12V, temos de considerar uma redução de aproximadamente 25% a 30% da capacitância, ou seja, 7uF por capacitor. Assim o banco de 5 capacitores tem uma capacitância de 35uF, que é o que se deseja para ripple de 1%.

### Aquecimento do capacitor com corrente

![typical](./datasheet/CL32B106KBJNFNE_temp.png)

Dado que estamos usando 5 capacitores, a corrente RMS de 462mA se divide entre os 5 capacitores, ou seja, cada capacitor tem uma corrente de 92.4mA por capacitor, assim o aumento de temperatura é desprezível, sendo menos do que 1 ºC.

### Resistência equivalente série (ESR)

![typical](./datasheet/CL32B106KBJNFNE_esr.png)

A resistência equivalente série (ESR) é de 7 mΩ por capacitor, ou seja, 1.4 mΩ para o banco de 5 capacitores. Assim, pode ser considerado desprezível para o projeto.

### Conclusões

Dado que se trata de um capacitor cerâmico de boa qualidade, esse capacitor vem a aumentar um pouco o custo do projeto, porém valendo a pena dado que irá aumentar muito a vida útil do conversor e a redução de volume do conversor. Assim proporcionando uma densidade de potência muito mais atrativa.

# Escolha da chave (IAUZ40N06S5N050)

![typical](./prints/switch.png)
![typical](./prints/switch_1.png)


Com base nos valores necessário de estresse para chave, foi escolhida a chave **IAUZ40N06S5N050** dado que se trata de uma chave barata e disponível na mouser e que atende os requisitos de corrente e tensão. E como mostrado a baixo não necessita de dissipador.

In [ ]:
# Converter
duty = 0.333
freq = 50e3

# Switch
i_s_rms = 4.8153
i_s_on = 9.11
v_ds_off = 36.0
switch_r_ds_on = 6e-3 # 6mΩ @ 100C

# Assuming gate driver can instantly turn on and off the switch
switch_tr = 1e-9
switch_tf = 4.6e-9

# Switch losses
switch_conduction_loss = switch_r_ds_on * (i_s_rms**2)
switch_commutation_loss = (freq / 2) * (switch_tr + switch_tf) * (v_ds_off * i_s_on)

switch_total_loss = switch_conduction_loss + switch_commutation_loss

print(f"Perdas de condução na chave: {switch_conduction_loss:.2f} W")
print(f"Perdas de comutação na chave: {switch_commutation_loss:.2f} W")
print(f"Perdas totais na chave: {switch_total_loss:.2f} W")

Perdas de condução na chave: 0.14 W
Perdas de comutação na chave: 0.05 W
Perdas totais na chave: 0.19 W


# Escolha do diodo (MBRD20150CT)

![typical](./prints/diode.png)
![typical](./prints/diode_1.png)

In [3]:
i_d_avg = 5.55
i_d_rms = 6.8122

# Assuming Tj = 25 since data sheet since 25 is the worse case
mbr_diode_v_threshold = 0.29
# Assumin Tj = 85 to get the RT closer to the operating point
mbr_diode_rt = 1.0 / 17.5421052 # Around 57mΩ for MBRD20150CT

mbr_diode_conduction_loss = mbr_diode_v_threshold * i_d_avg + mbr_diode_rt * (i_d_rms*2)
# Disconsider it since it is a Schottky diode
mbr_diode_commutation_loss = 0.0

mbr_diode_total_loss = mbr_diode_conduction_loss + mbr_diode_commutation_loss

print(f"Perdas de condução no diodo: {mbr_diode_conduction_loss:.2f} W")
print(f"Perdas de comutação no diodo: {mbr_diode_commutation_loss:.2f} W")
print(f"Perdas totais no diodo: {mbr_diode_total_loss:.2f} W")

Perdas de condução no diodo: 2.39 W
Perdas de comutação no diodo: 0.00 W
Perdas totais no diodo: 2.39 W


# Cálculo de dissipadores para validação

In [4]:
# Disconsidering loses on switch since are negligenciable IAUZ40N06S5N050

# MBRD20150CT
mbr_diode_r_junction_case = 6.0
mbr_diode_temp_max = 175.0 * 0.8 # Assuming 20% tolerance

# Assuming 70C ambient temperature
ambient_temp = 70.0

print(f"============ Calculo das perdas totais para o diodo MBRD20150CT ============")

diode_temp_sink_max = mbr_diode_temp_max - (mbr_diode_r_junction_case * mbr_diode_total_loss)

print(f"Temperatura maxima no dissipador para o diodo (MBRD20150CT): {diode_temp_sink_max:.2f} C")
print(f"Perdas totais: {mbr_diode_total_loss:.2f} W")

heatsink_r_max = (diode_temp_sink_max - ambient_temp) / mbr_diode_total_loss

print(f"Rth maximo do dissipador: {heatsink_r_max:.2f} C/W")


============ Calculo das perdas totais para o diodo MBRD20150CT ============
Temperatura maxima no dissipador para o diodo (MBRD20150CT): 125.68 C
Perdas totais: 2.39 W
Rth maximo do dissipador: 23.34 C/W


# Validação sobre a necessidade de dissipadores

Como cálculado, será negligenciado as perdas na chave tendo em vista que são despresíveis se comparada ao diodo e que potencialmente podem serem dissipadas pela própria placa devido a serem muito pequenas.

Assim sendo, o diodo é o único que possui um nível de perdas a serem consideradas. Conforme os cálculosm, o Rth do diodo é de 23.34 C/W, assim sendo, conforme seu datasheet:

![typical](./prints/diode_loss.png)

Não seria necessário um dissipador, lembrando que apesar de os valores de 22 C/W e 23.34 C/W serem bem próximos, tem de se considerar que ja há uma tolerância de 20% nos cálculos, assim sendo não seria necessario um dissipador.

## Recomendação extra

Como nota extra, caso a empresa que desenvolvendo o conversor deseje aumentar a vida útil ou operar em ambientes com temperaturas mais hostis, recomenda-se o uso de um dissipador pequeno para o conversor.

### HS 1509

Por exemplo um caso que pode ser utilizado é o HS 1509, que seria mais que suficiente tendo em vista que o conversor normal sem dissipador ja opera corretamente.

![typical](./datasheet/hs1509.png)